compress/compress.go

```diff
@@ -66,11 +66,9 @@ type Compressor struct {
 
 func NewCompressor(ctx context.Context, logPrefix, outputFile, tmpDir string, minPatternScore uint64, workers int) (*Compressor, error) {
 	dir, fileName := filepath.Split(outputFile)
-	ext := filepath.Ext(outputFile)
 	tmpOutFilePath := filepath.Join(dir, fileName) + ".tmp"
-	datFilePath := filepath.Join(tmpDir, fileName[:len(ext)]) + ".dat"
 
-	datFile, err := NewUncompressedFile(datFilePath)
+	datFile, err := NewUncompressedFile(outputFile)
```

Comment:
UncompressedFile - it's intermediate .dat file, outputFile it's final .seg file. 

tmpOutFilePath - it's ".seg.tmp" file which will be renamed to .seg file if everything succeed. It allow atomically create .seg file (downloader will not see partially ready/ non-ready .seg files). I didn't create ".seg.tmp" file in tmpDir, because I think tmpDir and snapsthoDir may be mounted to different drives - in this case os.Rename will fail.

Reply:
Ok, I see, thank you. I did not think about this